netket/graph/custom_graph.py

```diff
@@ -0,0 +1,99 @@
+from abstract_graph import PyAbstractGraph
+
+import numpy as _np
+import networkx as _nx
+
+class PyCustomGraph(PyAbstractGraph):
+	r"""A custom graph, specified by a list of edges and optionally colors."""
+
+	def __init__(self, edges=[]):
+		"""
+		Constructs a new graph given a list of edges.
+
+		Args:
+		    edges: If `edges` has elements of type `Tuple[int, int]` it is treated
+		        as a list of edges. Then each element `(i, j)` means a connection
+		        between sites `i` and `j`. It is assumed that `0 <= i <= j`. Also,
```

Comment:
does networkx assume that i <= j ?

Reply:
It doesn't, but it orders edges lexicographically. so that `g = nx.MultiGraph([[0,1],[2,1]])` is the same as `g = nx.MultiGraph([[0,1],[1,2]]) `

Reply:
Then we should remove the comment about that assumption, if we do not actually require sorted pairs. (Which can be slightly annoying with PBC.)